hail/src/main/scala/is/hail/expr/ir/TableWriter.scala

```diff
@@ -158,17 +164,23 @@ case class TableNativeWriter(
   }
 }
 
-case class PartitionNativeWriter(spec: AbstractTypedCodecSpec, partPrefix: String, index: Option[(String, PStruct)], localDir: Option[String]) extends PartitionWriter {
+case class PartitionNativeWriter(spec: AbstractTypedCodecSpec, optKeyType: Option[TStruct], partPrefix: String, index: Option[(String, PStruct)], localDir: Option[String]) extends PartitionWriter {
```

Comment:
can the key just be an `IndexedSeq[String]`?

Reply:
So that would then be indistinguishable from keying a table by no fields. If I have a table with a single row that has an empty key, should we say it's distinctly keyed? I guess it doesn't really matter since you can't join on such a thing.

Yeah, I'll change it to IndexedSeq[String]

Reply:
I think an unkeyed table is a table keyed by no fields, yes. Can't join on that so distinctness doesn't matter.